### PR TITLE
#25665 -- deprecate public getters/setters for properties of `GEOSGeometry` and its subclasses

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/25665

### CODE REVIEW

File: django/contrib/gis/geos/geometry.py [hunk 1]
```diff
@@ -364,10 +367,24 @@ def get_srid(self):
...
-    def set_srid(self, srid):
+    @srid.setter
+    def srid(self, srid):
         "Sets the SRID for the geometry."
         capi.geos_set_srid(self.ptr, 0 if srid is None else srid)
-    srid = property(get_srid, set_srid)
+
+    def get_srid(self):
+        warnings.warn(
+            "`get_srid` is deprecated, use `srid` property instead.",
```
Comment (timgraham) [inline django/contrib/gis/geos/geometry.py]: get_srid()
use the

File: django/contrib/gis/geos/point.py [hunk 2]
```diff
@@ -95,50 +97,104 @@ def _get_single_external(self, index):
...
+    def get_z(self):
+        warnings.warn(
+            "`get_z` is deprecated, use `z` property instead.",
+            RemovedInDjango20Warning, 2
+        )
+        return self.z
+
+    def set_z(self, value):
+        warnings.warn(
+            "`set_x` is deprecated, use `z` property instead.",
```
Comment (timgraham) [inline django/contrib/gis/geos/point.py]: set_z()

File: docs/releases/1.10.txt [hunk 3]
```diff
@@ -323,6 +323,20 @@ added in Django 1.9::
 
 This prevents confusion about an assignment resulting in an implicit save.
 
+:mod:`django.contrib.gis`
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (timgraham) [inline docs/releases/1.10.txt]: Add blank line after heading

File: docs/releases/1.10.txt [hunk 4]
```diff
@@ -323,6 +323,20 @@ added in Django 1.9::
...
+* ``get_srid`` and ``set_srid`` methods of
```
Comment (timgraham) [inline docs/releases/1.10.txt]: "The get_srid() and ... in favor of the ..." (similar comments for other notes). Also please add notes to docs/internals/deprecation.txt.

**Comment by sir-sigurd** [comment]
@timgraham 
Fixed, thanks for review!

**Comment by timgraham** [comment]
merged in b7177cc2a422a0183c8f2d56eceb6b9323c6f959, 7803f429a4d435623cb7b91dd324d3aadad87380, and 7a452c5ce295679307bd81dd9b5f37b3cf762acf, thanks!

**Pull request closed without merging**